### PR TITLE
INIT-1530 Snapshot syncing and handling incoming requests documentation improvements

### PR DESCRIPTION
**Pull request by tomgarner**
**Note: this is a public repo PR**

- adds request for manual resync config/availability actions for practice staff
- adds request to trigger a resync of config/availability if moving out of a consumer-software-based integration disabled configuration
- updates handling incoming requests section to mention that consumers should still poll periodically if using the webhook method to avoid missed requests
- mention that an integration might not be approved to go live if test scenarios mentioned in the suggested testing document are not accounted for

### CODE REVIEW

File: docs/quick-start-guide.md [hunk 1]
```diff
@@ -385,9 +398,13 @@ This will give you a result like so:
...
 
-### Polling
+### Polling method
+
+This method involves periodically polling our requests endpoint that will then tell you which other request endpoints require resolving.
+
+If you are not also implementing the webhook subscription method, this endpoint should be called frequently to ensure that requests are dealt with in a timely fashion.
 
-The alternative to primarily resolving requests via webhook payloads is to periodically poll our requests convenience endpoint that will then tell you which other request endpoints require resolving.
+If you are implementing the webhook subscription method, you should still poll this endpoint periodically on a less frequent basis. This practice helps in identifying any missed requests caused by transient network issues or the integration being temporarily disabled within your software.
```
Comment (soulsnake) [inline docs/quick-start-guide.md:407]:
```suggestion
If you are implementing the webhook subscription method, you should still poll this endpoint periodically on a less frequent basis. This practice helps in identifying any missed requests caused by transient network issues or the integration being temporarily disabled within your software. Polling **GET /requests** endpoint every hour at least is recommended in this case.
```

File: docs/quick-start-guide.md [hunk 2]
```diff
@@ -385,9 +398,13 @@ This will give you a result like so:
 
 This feature enables you to identify instances where the webhook has encountered multiple failures. In this particular scenario, it attempted to execute the task three times, and the log provided pertains to the last failed attempt.
 
-### Polling
+### Polling method
+
+This method involves periodically polling our requests endpoint that will then tell you which other request endpoints require resolving.
+
+If you are not also implementing the webhook subscription method, this endpoint should be called frequently to ensure that requests are dealt with in a timely fashion.
```
[the remark this answers is not in the document]
Comment (tomgarner) [reply docs/quick-start-guide.md]: Updated and just added to prior sentence - the requests endpoint is mentioned earlier in this section

> If you are not also implementing the webhook subscription method, this endpoint should be called at least once every five seconds to ensure that requests are dealt with in a timely fashion.

File: docs/quick-start-guide.md (hunk 1 above)
Comment (tomgarner) [reply docs/quick-start-guide.md]: Updated as:

> If you are implementing the webhook subscription method, you should still poll this endpoint at least once an hour. This practice helps in identifying any missed requests caused by transient network issues or the integration being temporarily disabled within your software.

**Comment by tomgarner** [comment]
After discussion with Mickael, added tests to call the requests endpoint in the following situations:
- during initial PMS account/location configuration sync (edge case from a previous integration)
- after the integration has been temporarily disabled due to software config when using the polling approach
- after the integration has been temporarily disabled due to software config when using the webhook approach

File: docs/quick-start-guide.md [hunk 3]
```diff
@@ -50,7 +50,11 @@ Once consent is granted, you will need to provide a way for the practice to prov
 
 Once a practice has granted you consent and has provided you their practice ID and practice key, some initial configuration must be supplied before attempting to sync available appointments and bookings.
 
-This configuration data should be transmitted both when a new Healthengine practice ID is associated with a PMS account or location and whenever there are updates to the corresponding data within the PMS.
+This configuration data should be transmitted when;
```
Comment (soulsnake) [inline docs/quick-start-guide.md:53]:
```suggestion
This configuration data should be transmitted when:
```

**Review by soulsnake — APPROVED** [review]

**Review by nathanblogs — APPROVED** [review]
